src/libraries/System.Private.CoreLib/gen/EventSourceGenerator.Parser.cs

```diff
@@ -4,119 +4,96 @@
 using System;
 using System.Collections.Generic;
 using System.Collections.Immutable;
-using System.Linq;
 using System.Security.Cryptography;
 using System.Text;
 using System.Threading;
 using Microsoft.CodeAnalysis;
-using Microsoft.CodeAnalysis.CSharp;
 using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.DotnetRuntime.Extensions;
 
 namespace Generators
 {
     public partial class EventSourceGenerator
     {
-        private static bool IsSyntaxTargetForGeneration(SyntaxNode node, CancellationToken cancellationToken) =>
-            node is ClassDeclarationSyntax { AttributeLists.Count: > 0 };
-
-        private static EventSourceClass? GetSemanticTargetForGeneration(GeneratorSyntaxContext context, CancellationToken cancellationToken)
+        private static EventSourceClass? GetSemanticTargetForGeneration(GeneratorAttributeSyntaxContext context, CancellationToken cancellationToken)
         {
-            const string EventSourceAutoGenerateAttribute = "System.Diagnostics.Tracing.EventSourceAutoGenerateAttribute";
             const string EventSourceAttribute = "System.Diagnostics.Tracing.EventSourceAttribute";
 
-            var classDef = (ClassDeclarationSyntax)context.Node;
-            SemanticModel sm = context.SemanticModel;
+            var classDef = (ClassDeclarationSyntax)context.TargetNode;
+            NamespaceDeclarationSyntax? ns = classDef.Parent as NamespaceDeclarationSyntax;
+            if (ns is null)
+            {
+                if (classDef.Parent is not CompilationUnitSyntax)
+                {
+                    // since this generator doesn't know how to generate a nested type...
+                    return null;
+                }
+            }
```

Comment:
moved this check up.  there was no point doing it in a loop since it would always be the same result.  and if it always caused us to continue, we'd never set `eventSourceClass`.  so we'd terminate the loop and would return `null`.  so this is effectively the same.

Reply:
note: this logic is wrong in that it can't handle:  `namespace n1 { namespace n2 { class Whatever { } } }`.  However, that bug existed before.  I didn't change it.  

A better check would likely be to do `if (classDef.Parent is TypeDeclarationSyntax)`